ttnn/cpp/ttnn/operations/ccl/barrier/barrier.cpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "barrier.hpp"
+
+namespace ttnn::operations::ccl {
+
+ttnn::Tensor BarrierOperation::invoke(
+    const Tensor& input_tensor,
+    const std::optional<ttnn::MemoryConfig>& memory_config,
+    ttnn::ccl::Topology topology)
+{
+    MemoryConfig out_memory_config = memory_config.value_or(input_tensor.memory_config());
+    return barrier
```

Comment:
I guess you call a function from barrier_op.hpp here, but it rather looks like a call to `constexpr auto barrier = ttnn...` this thing.

Reply:
I agree that the function would likely be better put in barrier.hpp but as I said in another comment, I really want to maintain consistency in the folder structure among ccl ops just to make navigating and debugging them easier in the future, and this is the way it is done in the others.

Reply:
I am saying there is a clash. You registered your op as `barrier` and you have a function called `barrier`, same namespace. This is not good.

Reply:
Oh I see what you mean, yeah will rename the namespace